Data not being displayed.

Data doesn't show up using MagicMirror v2.15.0 (fresh install). Clock Module and MMM-YNAB are the only modules registered. log seems to indicate API is working. I defined 1 category; it's in the "list" output, but the Mirror Display does not show anything other than the Clock module. I've tried changing the position attribute to no effect.

Hi. Thanks for taking a look at this so quickly.

This is what I get at startup now:

> magicmirror@2.15.0 start
> DISPLAY="${DISPLAY:=:0}" ./node_modules/.bin/electron js/electron.js

[23.04.2021 08:46.59.519] [LOG]   Starting MagicMirror: v2.15.0
[23.04.2021 08:46.59.521] [LOG]   Loading config ...
[23.04.2021 08:46.59.523] [LOG]   Loading module helpers ...
[23.04.2021 08:46.59.523] [LOG]   No helper found for module: alert.
[23.04.2021 08:46.59.574] [LOG]   Initializing new module helper ...
[23.04.2021 08:46.59.575] [LOG]   Module helper loaded: updatenotification
[23.04.2021 08:46.59.575] [LOG]   No helper found for module: clock.
[23.04.2021 08:46.59.578] [LOG]   Loading module helpers ...
[23.04.2021 08:46.59.579] [LOG]   No helper found for module: alert.
[23.04.2021 08:46.59.579] [LOG]   Initializing new module helper ...
[23.04.2021 08:46.59.579] [LOG]   Module helper loaded: updatenotification
[23.04.2021 08:46.59.579] [LOG]   No helper found for module: clock.
[23.04.2021 08:46.59.718] [LOG]   Launching application.
[23.04.2021 08:47.11.487] [LOG]   Shutting down server...
rolando@rolando-mm:~/MagicMirror$ npm start >> test.txt
[23.04.2021 08:47.37.928] [ERROR] WARNING! Could not load config file. Starting with default configuration. Error found: Error: Cannot find module '/home/rolando/MagicMirror/modules/MMM-YNAB/node_modules/ynab/index.js'. Please verify that the package.json has a valid "main" entry
[23.04.2021 08:47.37.932] [ERROR] App threw an error during load
[23.04.2021 08:47.37.933] [ERROR] Error: Cannot find module '/home/rolando/MagicMirror/modules/MMM-YNAB/node_modules/ynab/index.js'. Please verify that the package.json has a valid "main" entry
    at tryPackage (internal/modules/cjs/loader.js:328:19)
    at Function.Module._findPath (internal/modules/cjs/loader.js:689:18)
    at Module._resolveFilename (internal/modules/cjs/loader.js:959:27)
    at Function.n._resolveFilename (electron/js2c/browser_init.js:249:921)
    at Function.Module._resolveFilename (/home/rolando/MagicMirror/node_modules/module-alias/index.js:49:29)
    at Module._load (internal/modules/cjs/loader.js:848:27)
    at Function.f._load (electron/js2c/asar_bundle.js:5:12738)
    at Module.require (internal/modules/cjs/loader.js:1032:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at Object.<anonymous> (/home/rolando/MagicMirror/modules/MMM-YNAB/node_helper.js:1:14)
[23.04.2021 08:47.37.933] [ERROR] Whoops! There was an uncaught exception...
[23.04.2021 08:47.37.935] [ERROR] Error: Cannot find module '/home/rolando/MagicMirror/modules/MMM-YNAB/node_modules/ynab/index.js'. Please verify that the package.json has a valid "main" entry
    at tryPackage (internal/modules/cjs/loader.js:328:19)
    at Function.Module._findPath (internal/modules/cjs/loader.js:689:18)
    at Module._resolveFilename (internal/modules/cjs/loader.js:959:27)
    at Function.n._resolveFilename (electron/js2c/browser_init.js:249:921)
    at Function.Module._resolveFilename (/home/rolando/MagicMirror/node_modules/module-alias/index.js:49:29)
    at Module._load (internal/modules/cjs/loader.js:848:27)
    at Function.f._load (electron/js2c/asar_bundle.js:5:12738)
    at Module.require (internal/modules/cjs/loader.js:1032:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at Object.<anonymous> (/home/rolando/MagicMirror/modules/MMM-YNAB/node_helper.js:1:14) {
  code: 'MODULE_NOT_FOUND',
  path: '/home/rolando/MagicMirror/modules/MMM-YNAB/node_modules/ynab/package.json',
  requestPath: 'ynab'
}
[23.04.2021 08:47.37.936] [ERROR] MagicMirror will not quit, but it might be a good idea to check why this happened. Maybe no internet connection?
[23.04.2021 08:47.37.936] [ERROR] If you think this really is an issue, please open an issue on GitHub: https://github.com/MichMich/MagicMirror/issues

Hi. Thanks again for such a quick response. I did run npm install         though. Here, I'll start from scratch:

rolando@rolando-mm:~/MagicMirror/modules$ git clone https://github.com/thesoftwarejedi/MMM-YNAB.git
Cloning into 'MMM-YNAB'...
remote: Enumerating objects: 33, done.
remote: Counting objects: 100% (33/33), done.
remote: Compressing objects: 100% (24/24), done.
remote: Total 33 (delta 16), reused 19 (delta 8), pack-reused 0
Unpacking objects: 100% (33/33), 18.45 KiB | 1.84 MiB/s, done.
rolando@rolando-mm:~/MagicMirror/modules$ cd MMM-YNAB/
rolando@rolando-mm:~/MagicMirror/modules/MMM-YNAB$ npm install

added 8 packages, and audited 9 packages in 975ms

3 low severity vulnerabilities

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
rolando@rolando-mm:~/MagicMirror/modules/MMM-YNAB$ npm audit fix --force
npm WARN using --force Recommended protections disabled.
npm WARN audit Updating ynab to 0.0.1,which is a SemVer major change.

removed 7 packages, changed 1 package, and audited 2 packages in 893ms

found 0 vulnerabilities
rolando@rolando-mm:~/MagicMirror$ npm start

> magicmirror@2.15.0 start
> DISPLAY="${DISPLAY:=:0}" ./node_modules/.bin/electron js/electron.js

[23.04.2021 14:21.04.971] [LOG]   Starting MagicMirror: v2.15.0
[23.04.2021 14:21.04.972] [LOG]   Loading config ...
[23.04.2021 14:21.04.974] [LOG]   Loading module helpers ...
[23.04.2021 14:21.04.975] [LOG]   No helper found for module: alert.
[23.04.2021 14:21.05.032] [LOG]   Initializing new module helper ...
[23.04.2021 14:21.05.032] [LOG]   Module helper loaded: updatenotification
[23.04.2021 14:21.05.032] [LOG]   No helper found for module: clock.
[23.04.2021 14:21.05.035] [ERROR] WARNING! Could not load config file. Starting with default configuration. Error found: Error: Cannot find module '/home/rolando/MagicMirror/modules/MMM-YNAB/node_modules/ynab/index.js'. Please verify that the package.json has a valid "main" entry
[23.04.2021 14:21.05.035] [LOG]   Loading module helpers ...
[23.04.2021 14:21.05.036] [LOG]   No helper found for module: alert.
[23.04.2021 14:21.05.036] [LOG]   Initializing new module helper ...
[23.04.2021 14:21.05.036] [LOG]   Module helper loaded: updatenotification
[23.04.2021 14:21.05.036] [LOG]   No helper found for module: clock.
[23.04.2021 14:21.05.037] [ERROR] App threw an error during load
[23.04.2021 14:21.05.038] [ERROR] Error: Cannot find module '/home/rolando/MagicMirror/modules/MMM-YNAB/node_modules/ynab/index.js'. Please verify that the package.json has a valid "main" entry
    at tryPackage (internal/modules/cjs/loader.js:328:19)
    at Function.Module._findPath (internal/modules/cjs/loader.js:689:18)
    at Module._resolveFilename (internal/modules/cjs/loader.js:959:27)
    at Function.n._resolveFilename (electron/js2c/browser_init.js:249:921)
    at Function.Module._resolveFilename (/home/rolando/MagicMirror/node_modules/module-alias/index.js:49:29)
    at Module._load (internal/modules/cjs/loader.js:848:27)
    at Function.f._load (electron/js2c/asar_bundle.js:5:12738)
    at Module.require (internal/modules/cjs/loader.js:1032:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at Object.<anonymous> (/home/rolando/MagicMirror/modules/MMM-YNAB/node_helper.js:1:14)
[23.04.2021 14:21.05.038] [ERROR] Whoops! There was an uncaught exception...
[23.04.2021 14:21.05.040] [ERROR] Error: Cannot find module '/home/rolando/MagicMirror/modules/MMM-YNAB/node_modules/ynab/index.js'. Please verify that the package.json has a valid "main" entry
    at tryPackage (internal/modules/cjs/loader.js:328:19)
    at Function.Module._findPath (internal/modules/cjs/loader.js:689:18)
    at Module._resolveFilename (internal/modules/cjs/loader.js:959:27)
    at Function.n._resolveFilename (electron/js2c/browser_init.js:249:921)
    at Function.Module._resolveFilename (/home/rolando/MagicMirror/node_modules/module-alias/index.js:49:29)
    at Module._load (internal/modules/cjs/loader.js:848:27)
    at Function.f._load (electron/js2c/asar_bundle.js:5:12738)
    at Module.require (internal/modules/cjs/loader.js:1032:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at Object.<anonymous> (/home/rolando/MagicMirror/modules/MMM-YNAB/node_helper.js:1:14) {
  code: 'MODULE_NOT_FOUND',
  path: '/home/rolando/MagicMirror/modules/MMM-YNAB/node_modules/ynab/package.json',
  requestPath: 'ynab'
}
[23.04.2021 14:21.05.041] [ERROR] MagicMirror will not quit, but it might be a good idea to check why this happened. Maybe no internet connection?
[23.04.2021 14:21.05.041] [ERROR] If you think this really is an issue, please open an issue on GitHub: https://github.com/MichMich/MagicMirror/issues
[23.04.2021 14:21.05.183] [LOG]   Launching application.
^C/home/rolando/MagicMirror/node_modules/electron/dist/electron exited with signal SIGINT

--- a/MMM-YNAB.js
+++ b/MMM-YNAB.js
@@ -1,20 +1,21 @@
 Module.register("MMM-YNAB", {
+    result: [],
     defaults: {
         token: "",
         categories: [ "Household", "Pets", "Grocery", "Lunch", "Kids Clothes", "Restaurants", "Spontaneous Fun" ]
     },
 
     start: function () {
-        this.sendSocketNotification('SET_CONFIG', this.config);
+        this.sendSocketNotification('YNAB_SET_CONFIG', this.config);
     },
 
     getDom: function () {
         var wrapper = document.createElement("div");
         wrapper.classList = ["xsmall"];
-	console.log('Data length: ' + this.data.items.length);
-        if (this.data.items && this.data.items.length > 0) {
-	    for (let i of this.data.items) {
-                wrapper.innerHTML = this.data.items.map(a => "<span class='ynab-name'>" + a.name + "</span><span class='ynab-balance'>$" + (a.balance/1000).toFixed(2) + "</span>").join('');
+        wrapper.innerHTML = "Loading YNAB";
+        if (this.result.items && this.result.items.length > 0) {
+            for (let i of this.result.items) {
+                wrapper.innerHTML = this.result.items.map(a => "<span class='ynab-name'>" + a.name + "</span><span class='ynab-balance'>$" + (a.balance/1000).toFixed(2) + "</span>").join('');
             }
         }
         return wrapper;
@@ -22,13 +23,13 @@ Module.register("MMM-YNAB", {
 
     socketNotificationReceived: function (notification, payload) {
         console.log("notification: " + notification);
-        console.log("payload: " + payload);
-        if (notification == "UPDATE") {
-            this.data = payload;
+        console.log("payload: " + JSON.stringify(payload));
+        if (notification == "YNAB_UPDATE") {
+            this.result = payload;
             this.updateDom(0);
         }
     },
-    
+
     getStyles: function() {
         return [
             this.file('MMM-YNAB.css')

--- a/node_helper.js
+++ b/node_helper.js
@@ -8,21 +8,16 @@ var interval;
 module.exports = NodeHelper.create({
 
 	socketNotificationReceived: function (noti, payload) {
-		console.log("Notification received: " + noti);
-		if (noti == "SET_CONFIG") {
+		if (noti == "YNAB_SET_CONFIG") {
 			this.initialize(payload);
 		}
 	},
 
 	initialize: function (payload) {
-		console.log("initialize");
 		config = payload;
 		var ynabAPI = new ynab.API(config.token);
 
-		console.log("created api");
-
 		ynabAPI.budgets.getBudgets().then(budgetsResponse => {
-			console.log("budgetsResponse: " + JSON.stringify(budgetsResponse));
 			ynabBudgetId = budgetsResponse.data.budgets[0].id;
 			this.updateBudget();
 			if (!interval) {
@@ -35,18 +30,13 @@ module.exports = NodeHelper.create({
 	},
 
 	updateBudget: function () {
-		console.log("updateBudget: " + ynabBudgetId);
 		var ynabAPI = new ynab.API(config.token);
-		console.log("created api");
 		ynabAPI.categories.getCategories(ynabBudgetId).then(categoriesResponse => {
-			console.log("got cats");
 			const map = [].concat(...Array.from(categoriesResponse.data.category_groups.map(a => Array.from(a.categories)))).reduce((map, o) => { map[o.name] = o; return map; }, new Map());
 			var list = config.categories.map(a => map[a]).filter(a => a != undefined);
-			console.log("list: " + JSON.stringify(list));
-			self.sendSocketNotification("UPDATE", {
+			self.sendSocketNotification("YNAB_UPDATE", {
 				items: list,
 			});
-			console.log("sent update");
 		}).catch(e => {
 			console.log("error: " + e);
 		});

